pkg/spdk/engine.go

```diff
@@ -861,20 +902,51 @@ func (e *Engine) ReplicaDelete(spdkClient *spdkclient.Client, replicaName, repli
 	return nil
 }
 
-const SnapshotOperationCreate = "snapshot-create"
-const SnapshotOperationDelete = "snapshot-delete"
+const (
+	SnapshotOperationCreate = "snapshot-create"
+	SnapshotOperationDelete = "snapshot-delete"
+	SnapshotOperationRevert = "snapshot-revert"
+)
+
+func (e *Engine) SnapshotCreate(spdkClient *spdkclient.Client, inputSnapshotName string) (snapshotName string, err error) {
+	return e.snapshotOperation(spdkClient, inputSnapshotName, SnapshotOperationCreate)
+}
 
-func (e *Engine) SnapshotCreate(snapshotName string) (res *spdkrpc.Engine, err error) {
-	return e.snapshotOperation(snapshotName, SnapshotOperationCreate)
+func (e *Engine) SnapshotDelete(spdkClient *spdkclient.Client, snapshotName string) (err error) {
+	_, err = e.snapshotOperation(spdkClient, snapshotName, SnapshotOperationDelete)
+	return err
 }
 
-func (e *Engine) SnapshotDelete(snapshotName string) (res *spdkrpc.Engine, err error) {
-	return e.snapshotOperation(snapshotName, SnapshotOperationDelete)
+func (e *Engine) SnapshotRevert(spdkClient *spdkclient.Client, snapshotName string) (err error) {
+	_, err = e.snapshotOperation(spdkClient, snapshotName, SnapshotOperationRevert)
+	return err
 }
 
-func (e *Engine) snapshotOperation(snapshotName, snapshotOp string) (res *spdkrpc.Engine, err error) {
+func (e *Engine) snapshotOperation(spdkClient *spdkclient.Client, inputSnapshotName, snapshotOp string) (snapshotName string, err error) {
```

Comment:
nit: `snapshotOp string`, better to use type alias like `snapshotOp SnapshotOperationType`